packages/nextjs/src/config/types.ts

```diff
@@ -48,18 +45,284 @@ export type NextConfigObject = {
+   * or Sentry instance other than sentry.io.
+   *
+   * This value can also be set via the `SENTRY_URL` environment variable.
+   *
+   * Defaults to https://sentry.io/, which is the correct value for SaaS customers.
+   */
+  sentryUrl?: string;
+
+  /**
+   * Headers added to every outgoing network request.
+   */
+  headers?: Record<string, string>;
+
+  /**
+   * If set to true, internal plugin errors and performance data will be sent to Sentry.
+   *
+   * At Sentry we like to use Sentry ourselves to deliver faster and more stable products.
+   * We're very careful of what we're sending. We won't collect anything other than error
+   * and high-level performance data. We will never collect your code or any details of the
+   * projects in which you're using this plugin.
+   *
+   * Defaults to `true`.
+   */
+  telemetry?: boolean;
+
+  /**
+   * Suppresses all Sentry SDK build logs.
+   *
+   * Defaults to `false`.
+   */
+  // TODO: Actually implement this for the non-plugin code.
+  silent?: boolean;
+
+  /**
+   * Prints additional debug information about the SDK and uploading source maps when building the application.
+   *
+   * Defaults to `false`.
+   */
+  // TODO: Actually implement this for the non-plugin code.
+  debug?: boolean;
+
   /**
-   * Override the SDK's default decision about whether or not to enable to the Sentry webpack plugin for server files.
-   * Note that `false` forces the plugin to be enabled, even in situations where it's not recommended.
+   * Options for source maps uploading.
    */
-  disableServerWebpackPlugin?: boolean;
+  sourcemaps?: {
+    /**
+     * Disable any functionality related to source maps upload.
+     */
+    disable?: boolean;
+
+    /**
+     * A glob or an array of globs that specifies the build artifacts that should be uploaded to Sentry.
+     *
+     * If this option is not specified, the plugin will try to upload all JavaScript files and source map files that are created during build.
+     *
+     * The globbing patterns follow the implementation of the `glob` package. (https://www.npmjs.com/package/glob)
+     *
+     * Use the `debug` option to print information about which files end up being uploaded.
+     */
+    assets?: string | string[];
+
+    /**
+     * A glob or an array of globs that specifies which build artifacts should not be uploaded to Sentry.
+     *
+     * Default: `[]`
+     *
+     * The globbing patterns follow the implementation of the `glob` package. (https://www.npmjs.com/package/glob)
+     *
+     * Use the `debug` option to print information about which files end up being uploaded.
+     */
+    ignore?: string | string[];
+
+    /**
+     * Toggle whether generated source maps within your Next.js build folder should be automatically deleted after being uploaded to Sentry.
+     *
+     * Defaults to `false`.
+     */
+    // TODO: Add this option
+    // deleteSourcemapsAfterUpload?: boolean;
```

Comment:
IMHO it's probably also fine to just use the already available `filesToDeleteAfterUpload` option instead but feel free to implement this, too. I believe we had this option before in NextJS, right?

Reply:
I will probably think a bit harder about what to do here when I get to it. The reason why such a utility option may be nice is because the dist folder of Next.js projects is configurable.